snappy/snap_local.go

```diff
@@ -57,87 +59,115 @@ func NewInstalledSnap(yamlPath string) (*Snap, error) {
 	return snap, nil
 }
 
+func completeInfo(info *snap.Info, manifest *diskManifest) {
+	// XXX: we actually should not trust snap.yaml name!
+
+	info.Developer = SideloadedDeveloper
+	// default for compat with older installs
+	info.Channel = "stable"
+	info.Size = -1
+
+	if manifest != nil {
+		info.Revision = manifest.Revision
+		info.Developer = manifest.Developer
+		info.Channel = manifest.Channel
+		// store edits win!
+		info.Description = manifest.Description
+		// XXX: Summary doesn't exist in the store yet anyway
```

Comment:
the branch has become quite big already, I'll do it in an immediate subsequent one